### PR TITLE
Add Support Ticket functionality

### PR DESCRIPTION
**Pull request by kngako**
Similar to the iOS support ticket functionality

### CODE REVIEW

**Comment by tomastiminskas** [comment]
@kngako Some changes needed (screen attached as reference): 
- Make Problem EditText taller and multiline as on iOS (keeping layout design as similar as possible)
- Implement the `GET /logs` endpoint to relay to get last logs from server and show them below the problem EditText with similar format as on iOS
- Add the "Copy Logs" button and make buttons style similar to iOS

```
GET /logs
Response: 
{
    success: true,
    response: logs
}
```

IMG_0765

**Comment by kngako** [comment]
Should the logs be automatically included in the email or the user is required to paste them into the `EditText`?

**Comment by tomastiminskas** [comment]
@kngako  Logs are automatically included in Email body if they are present in the view (request to get logs could have failed)

https://github.com/stakwork/sphinx-ios/blob/0aa456c8ef74ac432cdae58777b4092909cfe99a/sphinx/Scenes/Left%20Menu/SupportViewController.swift#L83

But on iOS we are not automatically sending the email but opening the Email app with To email and body filled. So user can always delete the logs from the email body. I'm not sure if we can do something similar on android. If not let's discuss the options we have.

File: sphinx/screens-detail/support-ticket/support-ticket/src/main/java/chat/sphinx/support_ticket/ui/SupportTicketFragment.kt [hunk 1]
```diff
@@ -21,19 +34,97 @@ internal class SupportTicketFragment: BaseFragment<
...
+    )
+    private val sendMessageButton: LayoutButtonSendMessageBinding by viewBinding(
+        LayoutButtonSendMessageBinding::bind, R.id.include_button_send_message
+    )
+    private val copyLogsButton: LayoutButtonCopyLogsBinding by viewBinding(
+        LayoutButtonCopyLogsBinding::bind, R.id.include_button_copy_logs
+    )
+    private val layoutLogsTextBinding: LayoutLogsTextBinding by viewBinding(
+        LayoutLogsTextBinding::bind, R.id.include_log_text
+    )
```
Comment (05nelsonm) [inline sphinx/screens-detail/support-ticket/support-ticket/src/main/java/chat/sphinx/support_ticket/ui/SupportTicketFragment.kt:51]: unnecessary overhead as they can all be retrieved/used via
```kotlin
binding.apply {
    includeSupportTicketHeader
    includeSupportTicketLayout
    // ...
}
```

File: sphinx/screens-detail/support-ticket/support-ticket/src/main/java/chat/sphinx/support_ticket/ui/SupportTicketViewModel.kt [hunk 2]
```diff
@@ -1,15 +1,123 @@
...
+): SideEffectViewModel<
+        Context,
+        SupportTicketSideEffect,
+        SupportTicketViewState>(dispatchers, SupportTicketViewState.Empty)
 {
+    private inner class LogsStateContainer: ViewStateContainer<SupportTicketViewState>(SupportTicketViewState.Empty)
+
+    private val logsStateContainer: ViewStateContainer<SupportTicketViewState> by lazy {
+        LogsStateContainer()
+    }
```
Comment (05nelsonm) [inline sphinx/screens-detail/support-ticket/support-ticket/src/main/java/chat/sphinx/support_ticket/ui/SupportTicketViewModel.kt:38]: `ViewStateConainer<SupportTicketViewState>` is being lazily initialized via the `BaseViewModel` with the initial value of what was passed in it's constructor (`SupportTicketViewState.EMPTY`). it is available from via `viewModel.viewStateContainer`

File: sphinx/screens-detail/support-ticket/support-ticket/src/main/java/chat/sphinx/support_ticket/ui/SupportTicketViewModel.kt [hunk 3]
```diff
@@ -1,15 +1,123 @@
...
+
+    @JvmSynthetic
+    internal fun getLogsViewStateFlow(): StateFlow<SupportTicketViewState> =
+        logsStateContainer.viewStateFlow
+
+    @JvmSynthetic
+    internal fun updateLogsSendViewState(viewState: SupportTicketViewState) {
+        logsStateContainer.updateViewState(viewState)
+    }
```
Comment (05nelsonm) [inline sphinx/screens-detail/support-ticket/support-ticket/src/main/java/chat/sphinx/support_ticket/ui/SupportTicketViewModel.kt:47]: can be removed, and replaced in the fragment as
```kotlin
viewModel.collect { viewState ->
    // ...
}
```

```kotlin
viewModel.updateViewState(<ViewState>)
```

See `BaseViewModel` extension functions.

**Review by 05nelsonm — APPROVED** [review]
Have a few fixes that I noted, primarily around utilizing the Fragment/ViewModel frameworks but no biggie. Looks great!

We are targeting today to push a release, so I'll be merging this and handling the fixes on `develop`.

**Pull request merged**